toponetx/datasets/graph.py

```diff
@@ -13,15 +15,15 @@
 __all__ = ["karate_club"]
 
 
-def karate_club(complex_type="simplicial", feat_dim=2):
+def karate_club(
+    complex_type: Optional[str] = "simplicial", feat_dim: Optional[str] = 2
```

Comment:
- `complex_type` is not optional. The correct typing annotation would be `Literal["simplicial", "cell"]`
- `feat_dim` is not optional and is not of type `str`. The correct typing annotation would be just `int`.

Remember: `Optional` just means that `None` is an accepted value. This is not the case here. It has nothing to do with whether the caller has to provide a value or not. That is entirely determined by whether the parameter has a default value or not.

Reply:
great, thanks for being patient

Reply:
in that case, I do not need to check for the correct string when using Literal["simplicial", "cell"]?

Reply:
You still should check for the value inside the function, it's not a big overhead. Type annotations are not enforced by Python itself.